### PR TITLE
Address nits from previous PR stack + bump version

### PR DESCRIPTION
**Pull request by danthe3rd**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #518
* #511
* #510
* #509
* #505
* #508
* #504
* #491
* #502
* #490
* #498
* #488
* #487
* #486

### CODE REVIEW

File: version.txt [hunk 1]
```diff
@@ -1 +1 @@
-0.0.14.dev
+0.0.15.dev
```
Comment (fmassa) [inline version.txt:1]: I would have preferred if we bumped the version in a follow-up PR

File: setup.py [hunk 2]
```diff
@@ -133,8 +133,6 @@ def get_flash_attention_extensions(cuda_version: int, extra_compile_args):
                 "nvcc": extra_compile_args.get("nvcc", [])
                 + [
                     "-O3",
-                    "-U__CUDA_NO_HALF_OPERATORS__",
-                    "-U__CUDA_NO_HALF_CONVERSIONS__",
```
Comment (fmassa) [inline setup.py:137]: So this wasn't needed in the end?

**Review by fmassa — APPROVED** [review]
LGTM, thanks!

Also, if we are bumping the version, we might need to bump it as well on pypi. cc @dianaml0

File: setup.py (hunk 2 above)
Comment (danthe3rd) [reply setup.py:137]: Nope, I just copy pasted it because it was in cutlass's build process I think. Good catch

**Pull request merged**